gen/field.go

```diff
@@ -161,6 +161,10 @@ func compileJSONTag(f *compile.FieldSpec, name string, opts ...string) *structta
 		t.Options = append(t.Options, "omitempty")
 	}
 
+	if f.Required {
+		t.Options = append(t.Options, "required")
```

Comment:
Will this continue to work if I was doing, `(go.tag = "json:\"required\"")`? Can you add a test for this?

(You may want to check `HasOption` like the previous block.)